### PR TITLE
feat(person-on-events): Revamp pagination to handle 'lost' personIDs

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

See https://github.com/PostHog/posthog/issues/11488#issuecomment-1227186305

~~This is WIP. But wanted to get a PoC out before fixing everything.~~

Note that this fixes pagination for pre- person-on-events too, whenever deletes occur.

Now, instead of relying on the filtered result to tell us whether to paginate or not, we use the raw query output. If the query output >= limit, then paginate; otherwise don't. Note that this happens before any filtering of orphaned/deleted persons.

I'm also thinking of sending the raw query count, which would allow us to determine when to show a banner stating 'this went wrong because of person-on-events, etc. etc.'

Link pending, but the modal now looks like this: (Retention fixes in a different PR)

image

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/api/action.py [hunk 1]
```diff
@@ -190,19 +190,18 @@ def people(self, request: request.Request, *args: Any, **kwargs: Any) -> Respons
...
         entity = get_target_entity(filter)
 
-        actors, serialized_actors = TrendsActors(team, entity, filter).get_actors()
+        actors, serialized_actors, raw_count = TrendsActors(team, entity, filter).get_actors()
 
         current_url = request.get_full_path()
         next_url: Optional[str] = request.get_full_path()
+        limit = filter.limit or 100
         offset = filter.offset
-        if len(actors) > 100 and next_url:
```
Comment (neilkakkar) [inline posthog/api/action.py:198]: This was basically just an optimistic query earlier: We'd query 200 people, and hope atleast 100 are valid, so we could paginate. Not great.

Got rid of this logic completely & using `filter.limit` to govern this now, just like rest of insight actors.

**Comment by neilkakkar** [comment]
Also cc: @benjackwhite , I notice you're creating V2 of the Persons modal & planning to run an experiment for it, from the looks of it 👀 . Just a heads up.

**Comment by neilkakkar** [comment]
cc: @tiina303 do you know which link I can point this modal to? (It needs to explain why persons might be lost)

File: posthog/api/person.py [hunk 2]
```diff
@@ -257,7 +257,7 @@ def funnel(self, request: request.Request, **kwargs) -> response.Response:
         if not results_package:
```
Comment (EDsCODE) [inline posthog/api/person.py:257]: Is this new pattern not applicable to the regular person list?

Comment (neilkakkar) [reply posthog/api/person.py:257]: .. What do you mean? 🤔

Comment (EDsCODE) [reply posthog/api/person.py:257]: Oh it only happens in one place but posthog/api/cohort.py line 217 we're still using the postgres result and not paginating based on clickhouse raw count result

**Comment by tiina303** [comment]
> do you know which link I can point this modal to? (It needs to explain why persons might be lost)

https://posthog.com/docs/how-posthog-works/queries#insights-counting-unique-persons - is the section for it

File: posthog/api/person.py (hunk 2 above)
Comment (neilkakkar) [reply posthog/api/person.py:253]: Nice catch, have deleted the `should_paginate` function and replaced all usage

**Comment by neilkakkar** [comment]
@EDsCODE @clarkus ping for review :)

**Review by EDsCODE — COMMENTED** [review]
Post merge approval ✅

**Pull request merged**